src/Resources/Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs

```diff
@@ -221,6 +229,34 @@ public class NewAzureRoleAssignmentCommand : ResourcesBaseCmdlet
 
         public override void ExecuteCmdlet()
         {
+            if (ParameterSetName == ParameterSet.InputFile)
+            {
+                string fileName = this.TryResolvePath(InputFile);
+                if (!(new FileInfo(fileName)).Exists)
+                {
+                    throw new PSArgumentException(string.Format("File {0} does not exist", fileName));
+                }
+
+                try
+                {
+                    PSRoleAssignment RoleAssignment = JsonConvert.DeserializeObject<PSRoleAssignment>(File.ReadAllText(fileName));
+
+                    this.ObjectId = RoleAssignment.ObjectId;
+                    this.ResourceType = RoleAssignment.ObjectType;
+                    this.Scope = RoleAssignment.Scope;
+                    Guid guid = Guid.Empty;
+                    Guid.TryParse(RoleAssignment.RoleDefinitionId,out guid);
+                    this.RoleDefinitionId = guid;
+                    this.Description = RoleAssignment.Description;
+                    this.Condition = RoleAssignment.Condition;
+                    this.ConditionVersion = RoleAssignment.ConditionVersion;
+                }
+                catch (JsonException)
+                {
+                    WriteVerbose("Deserializing the input role definition failed.");
+                    throw;
```

Comment:
I suppose we should throw meaningful error message to the user.
A verbose log is by default not displayed.